core/kernel/dt_driver.c

```diff
@@ -256,6 +256,9 @@ static TEE_Result device_from_provider_prop(struct dt_driver_provider *prv,
 	struct dt_pargs *pargs = NULL;
 	unsigned int n = 0;
 
+	if (!prop)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
This can't be correct since `device_from_provider_prop()` is called with `prop == NULL` by `dt_driver_device_from_parent()`. So `NULL` has to be a valid case I suppose.

Reply:
Coverity is unhappy about the potential use of `prop` being in NULL in the  `fdt32_to_cpu()` function below.

Reply:
`assert(!prv->provider_cells || prop)`?

Reply:
OK but returning early with an error status will probably not work, if it would then it means we could replace `return device_from_provider_prop(prv, fdt, nodeoffset, NULL, device_ref);` by `return TEE_ERROR_BAD_PARAMETERS;`  in `dt_driver_device_from_parent()`.
Perhaps we assume `prop` is non-NULL as soon as `prv->provider_cells` is non-zero? Then how about:
```C
for (n = 0; n < prv->provider_cells; n++) {
		assert(prop);  /* Or "if (prop)"? */
		pargs->args[n] = fdt32_to_cpu(prop[n]);
}
```

Reply:
I'm fine with the latest.

Reply:
Not addressed.

Reply:
Sorry about that ..